pkg/servicescm/servicescm.go

```diff
@@ -278,6 +283,48 @@ func (s *Service) hasDependency(possibleDependencies []Service) bool {
 	return false
 }
 
+// hasCycle detects cycles between any of the given services by traversing the resulting dependency graph.
+// Wrapper for Service.hasCycle to handle disconnected graphs
```

Comment:
Can you add more color around the "disconnected graphs" case?

Reply:
Consider 3 services: A, B, and C
- A has no dependencies, B depends on C, C depends on B. 
- The resulting graph has 2 distinct, disconnected components: one is {A} and the other is {B,C}

As such, we cannot chose any node to call Service.hasCycles() on. If we simply did A.hasCycles(), it would return false despite the B<-->C cycle. So we need to call the method once on each subgraph.